zerver/worker/queue_processors.py

```diff
@@ -470,47 +470,12 @@ def get_state_handler(self):
         # type: () -> StateHandler
         return StateHandler()
 
-    def remove_leading_pattern(self, pattern, content):
-        # type: (str, str) -> Optional[str]
-        """
-        This function attempts to match and remove the pattern from the
-        beginning of the content.  The return value is the removal result if
-        there is a match, or None if there is not a match.
-        """
-        leading_pattern = re.compile(r'^' + pattern)
-        match = leading_pattern.match(content)
-        if match:
-            return content[len(match.group()):]
-        else:
-            return None
-
-    # TODO: Consolidate this with the code in bot_lib.py
-    def remove_leading_mention_if_necessary(self, message, user_profile):
-        # type: (Dict[str, Any], UserProfile) -> None
-        """
-        If the embedded bot is the leading @mention, then this function removes
-        the leading @mention from the message content (note that spaces after
-        the @mention also get stripped).  Otherwise, it leaves the message
-        unchanged.
-        """
-        mention_patterns = [
-            r'@({0})'.format(user_profile.full_name),
```

Comment:
Hmmm, https://github.com/zulip/zulip/blob/d5704478508205cdf39c941e9167fc51bb2b4026/api/bots_api/bot_lib.py#L157 covers fewer cases than the regexp here: the r`@({0})` without the asterisks for short_name. If this is resolved, LGTM!